c10/xpu/test/impl/XPUCachingAllocatorTest.cpp

```diff
@@ -0,0 +1,49 @@
+#include <gtest/gtest.h>
+
+#include <c10/xpu/XPUCachingAllocator.h>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+bool has_xpu() {
+  return c10::xpu::device_count() > 0;
+}
+
+TEST(XPUCachingAllocatorTest, GetXPUAllocator) {
+  auto* allocator = c10::xpu::XPUCachingAllocator::get();
+
+  auto _500mb = 500 * 1024 * 1024;
+  auto buffer = allocator->allocate(_500mb);
+
+  auto* xpu_allocator = c10::GetAllocator(buffer.device().type());
+  ASSERT_EQ_XPU(allocator, xpu_allocator);
+}
+
+TEST(XPUCachingAllocatorTest, DeviceCachingAllocate) {
+  auto* allocator = c10::xpu::XPUCachingAllocator::get();
+  auto _10mb = 10 * 1024 * 1024;
+  auto buffer = allocator->allocate(_10mb);
+  void* ptr0 = buffer.get();
+  // tmp is not allocated via device caching allocator.
+  void* tmp = sycl::aligned_alloc_device(
+      512, _10mb, c10::xpu::get_raw_device(0), c10::xpu::get_device_context());
+  void* ptr1 = c10::xpu::XPUCachingAllocator::raw_alloc(_10mb);
+  // ptr0 and ptr1 should be on the same block.
+  auto diff = static_cast<char*>(ptr1) - static_cast<char*>(ptr0);
```

Comment:
Since this is very implementation specific, I would suggest you add a comment here about what assumptions you make here on the implementation.

Reply:
OK, rich comments.